src/ui/MLFacet/MLFacet.ts

```diff
@@ -143,32 +148,29 @@ export class MLFacet extends Component {
     this.bind.onQueryState(MODEL_EVENTS.CHANGE, undefined, this.handleQueryStateChanged);
   }
 
-  private handleDoneBuildingQuery(data: IDoneBuildingQueryEventArgs) {
-    // TODO: add facets attribute to the query here
-    // TODO: remove update of the mock
-    if (!this.mockedSavedValues) {
-      this.mockedSavedValues = [
-        { value: 'test 1', selected: false, numberOfResults: 847324 },
-        { value: 'test 2', selected: false, numberOfResults: 1 },
-        { value: 'test 3', selected: false, numberOfResults: 13 },
-        { value: 'test 4', selected: false, numberOfResults: 13134 },
-        { value: 'test 5', selected: false, numberOfResults: 2223 }
-      ] as IMLFacetValue[];
-      return;
-    }
+  private initMLFacetQueryController() {
+    this.mLFacetQueryController = new MLFacetQueryController(this);
+  }
 
-    this.mockedSavedValues = this.values.allFacetValues;
+  private handleDoneBuildingQuery(data: IDoneBuildingQueryEventArgs) {
+    Assert.exists(data);
+    Assert.exists(data.queryBuilder);
+    const queryBuilder = data.queryBuilder;
+    this.mLFacetQueryController.putFacetIntoQueryBuilder(queryBuilder);
   }
 
   private handleQuerySuccess(data: IQuerySuccessEventArgs) {
-    // TODO: mock response elsewhere
-    data.results.facets = this.mockedSavedValues;
-
     if (Utils.isNullOrUndefined(data.results.facets)) {
       return this.notImplementedError();
     }
 
-    this.onQueryResponse(data.results.facets);
+    const facetResponse = findWhere(data.results.facets, { field: this.fieldName });
```

Comment:
You're right it's not reliable at all.
This is not final, for now it's sufficient to see the values, but still have to come up with a manager that will be able to adapt the order of the visible facet.
I will go for now with the sort of unique ID, but it will be in a future PR.